jbmc/src/java_bytecode/java_object_factory.cpp

```diff
@@ -249,9 +249,11 @@ void java_object_factoryt::gen_pointer_target_init(
   PRECONDITION(expr.type().id() == ID_pointer);
   PRECONDITION(update_in_place != update_in_placet::MAY_UPDATE_IN_PLACE);
 
-  if(target_type.id() == ID_struct)
+  const typet &followed_target_type = ns.follow(target_type);
+
+  if(followed_target_type.id() == ID_struct)
```

Comment:
:pick: Prefer not directly accessing the id:

```C++
if(const auto target_class_type = try_type_dynamic_cast<java_class_type>(followed_target_type))
```

:eagle:

Reply:
Has the advantage that if we choose to tag java_class_type to differentiate from other struct types, will automatically get checked here.

Reply:
I'll let the `id()` / `dynamic_cast` war work itself out in the other thread before doing this on existing code :)